WooCommerce/UITestsFoundation/Screens/Products/ProductsScreen.swift

```diff
@@ -59,14 +48,17 @@ public final class ProductsScreen: ScreenObject {
         return try SingleProductScreen()
     }
 
-    public func verifyProductListOnProductsScreen(count: Int, name: String, status: String) throws -> Self {
-        XCTAssertTrue(try app.getTextVisibilityCount(text: name) == 1, "Product name does not exist!")
-        XCTAssertTrue(app.tables.cells.count == count, "Expecting \(count) products, got \(app.tables.cells.count) instead!")
+    func verifyStockStatusForProduct(name: String, status: String) throws -> Bool {
+        let namePredicate = NSPredicate(format: "identifier == %@", name)
+        let statusPredicate = NSPredicate(format: "label CONTAINS[c] %@", status)
 
-        /// TODO: Add AccessibilityIdentifiers to product list cell elements to be able to get stock status label value and update this assert.
-        /// Currently mock response API has "in stock" which is set to the first product repeated 3 times, so current assert will check that it appears 3 times
-        XCTAssertTrue(try app.getTextVisibilityCount(text: status) == 3, "Stock status does not exist!")
+        return app.tables.cells.containing(namePredicate).children(matching: .staticText).element(matching: statusPredicate).firstMatch.exists
+    }
 
+    public func verifyProductListOnProductsScreen(products: [ProductData]) throws -> Self {
+        XCTAssertTrue(try app.getTextVisibilityCount(text: products[0].name) == 1, "Product name does not exist!")
+        XCTAssertTrue(app.tables.cells.count == products.count, "Expecting \(products.count) products, got \(app.tables.cells.count) instead!")
```

Comment:
Just a thought that maybe `XCTAssertEqual` would be more suitable here.

```suggestion
        XCTAssertEqual(products.count, app.tables.cells.count, "Expecting \(products.count) products, got \(app.tables.cells.count) instead!")
```

You can also note that I swapped the first and second arguments, because the usual order in AssertEqual is `expected, actual, optional_message`